atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CQLKeyValueService.java

```diff
@@ -1114,8 +1114,8 @@ String getFullTableName(String tableName) {
     @Override
     public void compactInternally(String tableName) {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(tableName), "tableName:[%s] should not be null or empty", tableName);
-        final CassandraKeyValueServiceConfig config = configManager.getConfig();
-        if(!compactionManager.isPresent() || !config.jmx().isPresent()){
+        CassandraKeyValueServiceConfig config = configManager.getConfig();
+        if (!compactionManager.isPresent() || !config.jmx().isPresent()) {
             log.warn("No compaction client was configured, but compact was called. If you actually want to clear deleted data immediately " +
```

Comment:
Done. Removed one condition check as well, since config.jmx().isPresent() check is performed inside CassandraJmxCompaction.createJmxCompactionManager().